src/v/redpanda/admin/api-doc/transform.json

```diff
@@ -0,0 +1,155 @@
+{
+  "apiVersion": "0.0.1",
+  "swaggerVersion": "1.2",
+  "basePath": "/v1",
+  "resourcePath": "/transform",
+  "apis": [
+    {
+      "path": "/v1/transform/deploy",
+      "operations": [
+        {
+          "method": "POST",
+          "summary": "Deploy a transform based on the provided metadata and wasm binary in the request payload. The payload should be a json `deploy_transform_request` object immediately followed by the wasm binary without any delimiters.",
+          "nickname": "deploy_transform",
+          "consumes": [
+            "application/json+wasm"
+          ],
+          "type": "void",
+          "produces": [
+            "application/json"
+          ]
+        }
+      ]
+    },
+    {
+      "path": "/v1/transform/delete",
+      "operations": [
+        {
+          "method": "POST",
+          "summary": "Delete a transform based on the provided name in the request payload.",
+          "type": "void",
+          "nickname": "delete_transform",
+          "consumes": [
+            "application/json"
+          ],
+          "produces": [
+            "application/json"
+          ]
+        }
+      ]
+    },
+    {
+      "path": "/v1/transform/list",
+      "operations": [
+        {
+          "method": "GET",
+          "summary": "List all transforms, the status report contains a cluster wide aggregated view of the live transform execution state.",
+          "nickname": "list_transforms",
+          "produces": [
+            "application/json"
+          ]
+        }
+      ]
+    }
+  ],
+  "models": {
+    "deploy_transform_request": {
+      "id": "deploy_transform_request",
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "input_topic": {
+          "type": "string"
+        },
+        "output_topics": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "environment": {
+          "type": "array",
+          "items": {
+            "type": "environment_variable"
+          },
+          "description": "The environment variable configuration for a transform"
+        }
+      }
+    },
+    "delete_transform_request": {
+      "id": "delete_transform_request",
+      "properties": {
+        "name": {
+          "type": "string"
+        }
+      }
+    },
```

Comment:
If the `delete_transform_request` is simply a string, then couldn't the request payload just take a string?
String is well-typed as well.

Reply:
This is more extensible if we want to add additional properties in the future without breaking changes. Examples are conditional deletes like delete at version X or something like that.

Reply:
That makes sense.